app/src/main/res/layout/activity_microscope_quality.xml

```diff
@@ -0,0 +1,70 @@
+<?xml version="1.0" encoding="utf-8"?>
+
+<androidx.constraintlayout.widget.ConstraintLayout xmlns:android="http://schemas.android.com/apk/res/android"
+    xmlns:app="http://schemas.android.com/apk/res-auto"
+    xmlns:tools="http://schemas.android.com/tools"
+    android:layout_width="match_parent"
+    android:layout_height="match_parent"
+    android:orientation="vertical"
+    tools:context=".ui.microscope_quality.MicroscopeQualityActivity">
+
+    <include
+        layout="@layout/toolbar"
+        app:layout_constraintStart_toStartOf="parent"
+        app:layout_constraintTop_toTopOf="parent" />
+
+    <androidx.constraintlayout.widget.ConstraintLayout
+        android:padding="16dp"
+        android:layout_height="0dp"
+        android:layout_width="match_parent"
+        app:layout_constraintTop_toBottomOf="@id/toolbar"
+        app:layout_constraintBottom_toTopOf="@id/sample_preparation_continue_button">
+
+        <androidx.appcompat.widget.SwitchCompat
+            android:id="@+id/microscope_quality_damaged_switch"
+            android:layout_width="match_parent"
+            android:layout_height="wrap_content"
+            android:text="@string/microscope_quality_damaged_title"
+            app:layout_constraintTop_toTopOf="parent" />
+
+        <TextView
+            android:id="@+id/microscope_quality_magnification_title"
+            style="@style/TextView.SectionTitle"
+            android:layout_width="wrap_content"
+            android:layout_height="wrap_content"
+            android:text="@string/microscope_quality_magnification_title"
+            app:layout_constraintEnd_toEndOf="parent"
+            app:layout_constraintHorizontal_bias="0.0"
+            app:layout_constraintStart_toStartOf="parent"
+            app:layout_constraintTop_toBottomOf="@id/microscope_quality_damaged_switch" />
+
+        <EditText
+            android:id="@+id/microscope_quality_magnification_edit_text"
+            android:layout_width="wrap_content"
+            android:layout_height="wrap_content"
+            android:ems="10"
+            android:inputType="number"
+            app:layout_constraintStart_toStartOf="@id/microscope_quality_magnification_title"
+            app:layout_constraintTop_toBottomOf="@id/microscope_quality_magnification_title" />
+
+        <TextView
+            android:id="@+id/microscope_quality_magnification_error"
+            android:layout_width="match_parent"
+            android:layout_height="wrap_content"
+            android:text="@string/microscope_quality_magnification_error"
+            android:visibility="gone"
+            app:layout_constraintStart_toStartOf="@id/microscope_quality_magnification_edit_text"
+            app:layout_constraintTop_toBottomOf="@id/microscope_quality_magnification_edit_text"
+            android:textColor="@color/colorError" />
```

Comment:
These three widgets can be changed for a [`TextInputLayout`](https://material.io/develop/android/components/text-fields/) to achieve the same functionality in a more interactive and Material-like way. What do you think, would it be worth using that instead?

Reply:
Great call, thanks!